### PR TITLE
parser for @supports at rule

### PR DESCRIPTION
**Pull request by ideadapt**
Hi

I try to give `@supports` a chance to be part of this great parser. Have a look at my code. I appreciate any feedback.

This PR addresses issue #93.

### CODE REVIEW

**Comment by cscott** [comment]
Looks pretty good.  Could you rebase & squash this?

**Comment by ideadapt** [comment]
Done. Thanks for immediate feedback.

File: .gitignore [hunk 1]
```diff
@@ -11,4 +11,5 @@ Desktop.ini
 .DS_Store
 
 # NPM files
-npm-debug.log
\ No newline at end of file
+npm-debug.log
+node_modules/
```
Comment (cscott) [inline .gitignore]: Could you revert this change?  We'll probably want to do this eventually, but it doesn't make sense as part of this patch.

File: src/css/Parser.js [hunk 2]
```diff
@@ -334,6 +338,138 @@ Parser.prototype = function(){
...
             },
 
+            _supports: function(emit){
+                /*
+                 * supports_rule
+                 *  : SUPPORTS_SYM S* supports_condition S* group_rule_body
+                 *  ;
+                 */
+                var tokenStream = this._tokenStream,
+                    token,
```
Comment (cscott) [inline src/css/Parser.js]: `token` is unused here.  Please delete.

File: src/css/Parser.js [hunk 3]
```diff
@@ -334,6 +338,138 @@ Parser.prototype = function(){
...
+                    } else {
+                        tokenStream.unget();
+                    }
+                } else{
+                    this._supports_condition_in_parens();
+                    this._readWhitespace();
+
+                    while(tokenStream.peek() === Tokens.IDENT){
+                        ident = tokenStream.LT(1).value.toLowerCase();
+                        if(ident == "and" || ident == "or"){
```
Comment (cscott) [inline src/css/Parser.js]: Please use `===` here instead of `==`.  (In both tests.)

**Comment by tophf** [comment]
Great! Should be added [inside @-moz-document loop](https://github.com/CSSLint/parser-lib/blob/5c8b47138a3429804d8b81d7c3716ea2466e3318/src/css/Parser.js#L955). Will you do it?
